service/src/main/kotlin/fi/espoo/evaka/dvv/DvvModificationsService.kt

```diff
@@ -0,0 +1,161 @@
+                if (addressDvvInfoGroup.changeAttribute.equals("LISATTY") || (
+                    addressDvvInfoGroup.changeAttribute.equals("MUUTETTU") && it.streetAddress.isNullOrEmpty()
+                    )
+                ) {
+                    logger.debug("Dvv modification for ${it.id}: address change, type: ${addressDvvInfoGroup.changeAttribute}")
+                    h.updatePersonFromVtj(
+                        it.copy(
+                            streetAddress = addressDvvInfoGroup.streetAddress(),
+                            postalCode = addressDvvInfoGroup.postalCode ?: "",
+                            postOffice = addressDvvInfoGroup.postOffice?.fi ?: ""
+                            // TODO: residence code
+                        )
+                    )
+                }
+            }
+        }
+    }
+
+    private fun handleCustodianLimitedDvvInfoGroup(ssn: String, custodianLimitedDvvInfoGroup: CustodianLimitedDvvInfoGroup) {
+        if (custodianLimitedDvvInfoGroup.changeAttribute == "LISATTY") {
+            val user = AuthenticatedUser.anonymous
+            personService.getOrCreatePerson(user, ExternalIdentifier.SSN.getInstance(ssn))?.let {
+                logger.debug("Dvv modification for ${it.id}: has a new custodian, refreshing all info from DVV")
+                fridgeFamilyService.doVTJRefresh(VTJRefresh(it.id, user.id))
+            }
+        }
+    }
+
+    private fun handleCaretakerLimitedDvvInfoGroup(caretakerLimitedDvvInfoGroup: CaretakerLimitedDvvInfoGroup) {
+        if (caretakerLimitedDvvInfoGroup.changeAttribute == "LISATTY") {
+            val user = AuthenticatedUser.anonymous
+            personService.getOrCreatePerson(user, ExternalIdentifier.SSN.getInstance(caretakerLimitedDvvInfoGroup.caretaker.ssn))?.let {
+                logger.debug("Dvv modification for ${it.id}: a new caretaker added, refreshing all info from DVV")
+                fridgeFamilyService.doVTJRefresh(VTJRefresh(it.id, user.id))
+            }
+        }
+    }
+
+    // private fun handlePersonNameDvvInfoGroup(ssn: String, personNameDvvInfoGroup: PersonNameDvvInfoGroup) {}
+
+    // private fun handlePersonNameChangeDvvInfoGroup(ssn: String, personNameChangeDvvInfoGroup: PersonNameChangeDvvInfoGroup) {}
+
+    fun getDvvModifications(ssns: List<String>): List<DvvModification> = jdbi.transaction { h ->
+        val token = getNextDvvModificationToken(h)
+        logger.debug("Fetching dvv modifications with $token")
+        dvvModificationsServiceClient.getModifications(token, ssns)?.let { dvvModificationsResponse ->
+            storeDvvModificationToken(h, token, dvvModificationsResponse.modificationToken, ssns.size, dvvModificationsResponse.modifications.size)
+            dvvModificationsResponse.modifications
+        } ?: emptyList()
+    }
+}
+
+// Forms a list of persons' ssn's that should be updated from DVV
+fun getPersonSsnsToUpdate(jdbi: Jdbi): List<String> = jdbi.handle { h ->
```

Comment:
This function seems to be a duplicate? There's another in DvvModificationsBatchRefreshService